### PR TITLE
Add base implementation that would make CI happy

### PR DESCRIPTION
**Pull request by yonatankarp**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -51,12 +51,16 @@ jobs:
...
-      - name: Test Local Action
+      - name: Test Display All Mutations
         id: test-action
         uses: ./
         with:
-          who-to-greet: GitHub
+          file-path: ./mutations.xml
+
+      - name: Test Display Survives Only
+        id: test-action
```
Comment (msmolev) [inline .github/workflows/ci.yml]: This line fails in linter because it has the same id `test-action` as line 55

Comment (yonatankarp) [reply .github/workflows/ci.yml]: 🤦 Thanks!

File: src/main.js [hunk 2]
```diff
@@ -1,28 +1,68 @@
...
+        for (const mutation of mutations) {
+          reportContent += `- ${mutation}\n`
+        }
+        reportContent += '\n'
+      }
+
+      // Write directly to the GitHub Step Summary using @actions/core
+      core.summary.addRaw(reportContent).write()
+    })
+  })
```
Comment (msmolev) [inline src/main.js:65]: Formatting is a bit off, so you probably want to do  `npx prettier --write`

Comment (yonatankarp) [reply src/main.js:65]: I assume that GitHub did it to be agnostic to the linter usage, but it took me a while to understand that actually needed to run

```shell
$ npm run format:write
```

**Comment by yonatankarp** [comment]
This PR was break down to multiple PRs as the score of it continued to change while trying to make things work.

**Pull request closed without merging**